src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_ConditionalExpressions.vb

```diff
@@ -236,12 +231,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
             End If
 
-            ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right is the default value of T
             If rewrittenLeft.Type.IsNullableType() AndAlso
-               rewrittenRight.IsDefaultValue() AndAlso
                rewrittenRight.Type.IsSameTypeIgnoringAll(rewrittenLeft.Type.GetNullableUnderlyingType()) _
             Then
-                Return NullableValueOrDefault(rewrittenLeft)
+                ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right is the default value of T
+                If rewrittenRight.IsDefaultValue() Then
+                    Return NullableValueOrDefault(rewrittenLeft)
+                End If
+
+                ' Optimize If(left, right) to left.GetValueOrDefault() when left is T? and right value is a constant of type T
+                If rewrittenRight.ConstantValueOpt IsNot Nothing OrElse
+                   (rewrittenRight.Kind = BoundKind.Local AndAlso Not DirectCast(rewrittenRight, BoundLocal).LocalSymbol.IsRef) OrElse
```

Comment:
>(rewrittenRight.Kind = BoundKind.Local AndAlso Not DirectCast(rewrittenRight, BoundLocal).LocalSymbol.IsRef) [](http://example.com/codeflow?start=19&length=108)

This isn't a request for a change. Just a note. This expression could be shortened to:
```
Not If(TryCast(rewrittenRight, BoundLocal)?.LocalSymbol.IsRef, True)
```